pm/sigverifier_test.go

```diff
@@ -0,0 +1,76 @@
+package pm
+
+import (
+	"fmt"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestVerify(t *testing.T) {
+	msg := []byte("foo")
+	personalMsg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", 32, msg)
+	personalHash := crypto.Keccak256([]byte(personalMsg))
+
+	senderPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sender := crypto.PubkeyToAddress(senderPrivKey.PublicKey)
+
+	senderSig, err := crypto.Sign(personalHash, senderPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signerPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signer := crypto.PubkeyToAddress(signerPrivKey.PublicKey)
+
+	signerSig, err := crypto.Sign(personalHash, signerPrivKey)
+
+	b := &stubBroker{
+		usedTickets:     make(map[ethcommon.Hash]bool),
+		approvedSigners: make(map[ethcommon.Address]bool),
+	}
+
+	sv := NewApprovedSigVerifier(b)
+
+	// Test invalid signature for non-approved signer
+	valid, err := sv.Verify(sender, signerSig, msg)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
I think that starting from this line forward you want to switch from `t.Fatal` to `t.Fail`, since I believe `Fatal` stops the test function on that line and does not proceed, versus `Fail` that allows the test to continue, therefore not masking the rest of the checks in this function.

wdyt?

Reply:
We also use `t.Error` elsewhere in the codebase to indicate "fail but continue"

Reply:
So what should be the consistent way to do it? `t.Error` or `t.Fail`?

Reply:
Sounds like `t.Error` is the choice then!